staging/src/k8s.io/apiserver/pkg/apis/audit/types.go

```diff
@@ -125,6 +125,12 @@ type Event struct {
 	RequestReceivedTimestamp metav1.MicroTime
 	// Time the request reached current audit stage.
 	StageTimestamp metav1.MicroTime
+
+	// Annotations is an unstructured key value map stored with an event. Mechanisms like RBAC and
+	// PSP will store admission information in Annotations, like name of the policy which admits
+	// the request.
+	// +optional
+	Annotations map[string]string
```

Comment:
@tallclair @crassirostris @ericchiang @soltysh this is about the API. Are we happy with this representation?

Reply:
I'm fine with this interface, but I remember someone was not happy with it

Reply:
We have plugin names (those which are also used in the apiserver flags). We could make this more structured using them if we concentrate on admission. We could render the whole chain here in the event.

RBAC would then be another step, probably with an explicit authorization field.

Reply:
I just spoke with @sttts and we agreed it would be nice to:
1. be able to decide which admission and rbac plugins are logged, so that would require additional rule changes
2. Authorization logging happens in `authorizingVisitor` which is perfect, but admission happens only in PSP, I'd imagine this should happen a level higher to be able to do what I said in 1. 

@tallclair @crassirostris @ericchiang @CaoShuFeng opinions?

Reply:
> be able to decide which admission and rbac plugins are logged, so that would require additional rule changes

This feels like something that should be resolved with filters, if we ever get around to implementing them. We don't currently allow fine grained selection of what's logged, so I think the current approach is consistent.

> Authorization logging happens in authorizingVisitor which is perfect, 

Only for RBAC, it's an implementation detail.

> but admission happens only in PSP, I'd imagine this should happen a level higher to be able to do what I said in 1.

I see this field as arbitrary metadata that plugins invoked in the serving path can add to. I think the authorization & admission interfaces should enable this on a case-by-case basis.

I'm happy with the current unstructured approach.